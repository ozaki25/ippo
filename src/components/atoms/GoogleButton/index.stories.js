import React from 'react';
import { action } from '@storybook/addon-actions';
import GoogleButton from '.';

export default {
  title: 'atoms/GoogleButton',
};

export const 通常パターン = () => <GoogleButton onClick={action('click')} />;
