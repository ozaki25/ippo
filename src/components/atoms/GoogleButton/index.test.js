import React from 'react';
import { snapshot } from 'src/test/helpers';
import GoogleButton from '.';

const props = () => ({
  onClick: jest.fn(),
});

snapshot('GoogleButton/nomal', <GoogleButton {...props({})} />);
