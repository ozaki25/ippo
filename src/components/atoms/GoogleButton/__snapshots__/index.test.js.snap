// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: GoogleButton/nomal 1`] = `
.c0 {
  cursor: pointer;
}

.c0:active {
  opacity: 0.4;
}

<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MockRouter
    createHref={[Function]}
    location=""
    params={Object {}}
    path=""
    push={[Function]}
    replace={[Function]}
    url=""
  >
    <GoogleButton
      onClick={[MockFunction]}
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "lastClassName": "c0",
              "rules": Array [
                "
  cursor: pointer;
  &:active {
    opacity: 0.4;
  }
",
              ],
            },
            "displayName": "GoogleButton",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          }
        }
        forwardedRef={null}
        onClick={[MockFunction]}
      >
        <GoogleButton
          className="c0"
          onClick={[MockFunction]}
        >
          <svg
            className="c0"
            height="3em"
            onClick={[MockFunction]}
            viewBox="0 0 193 48"
            width="12em"
          >
            <defs>
              <rect
                height={40}
                id="btn_google_signin_dark_normal_web_svg__b"
                rx={2}
                width={185}
                x={0}
                y={0}
              />
              <filter
                filterUnits="objectBoundingBox"
                height="112.5%"
                id="btn_google_signin_dark_normal_web_svg__a"
                width="102.2%"
                x="-1.1%"
                y="-5%"
              >
                <feOffset
                  dy={1}
                  in="SourceAlpha"
                  result="shadowOffsetOuter1"
                />
                <feGaussianBlur
                  in="shadowOffsetOuter1"
                  result="shadowBlurOuter1"
                  stdDeviation={0.5}
                />
                <feColorMatrix
                  in="shadowBlurOuter1"
                  result="shadowMatrixOuter1"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.24 0"
                />
                <feOffset
                  in="SourceAlpha"
                  result="shadowOffsetOuter2"
                />
                <feGaussianBlur
                  in="shadowOffsetOuter2"
                  result="shadowBlurOuter2"
                  stdDeviation={0.5}
                />
                <feColorMatrix
                  in="shadowBlurOuter2"
                  result="shadowMatrixOuter2"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0.12 0"
                />
                <feMerge>
                  <feMergeNode
                    in="shadowMatrixOuter1"
                  />
                  <feMergeNode
                    in="shadowMatrixOuter2"
                  />
                </feMerge>
              </filter>
            </defs>
            <g
              fill="none"
              fillRule="evenodd"
            >
              <g
                transform="translate(4 4)"
              >
                <use
                  fill="#000"
                  filter="url(#btn_google_signin_dark_normal_web_svg__a)"
                  xlinkHref="#btn_google_signin_dark_normal_web_svg__b"
                />
                <use
                  fill="#4285F4"
                  xlinkHref="#btn_google_signin_dark_normal_web_svg__b"
                />
              </g>
              <text
                fill="#FFF"
                fontFamily="Helvetica"
                fontSize={14}
                letterSpacing={0.219}
              >
                <tspan
                  x={54}
                  y={30}
                >
                  Sign in with Google
                </tspan>
              </text>
              <rect
                fill="#FFF"
                height={38}
                rx={1}
                width={38}
                x={5}
                y={5}
              />
              <path
                d="M32.64 24.205c0-.639-.057-1.252-.164-1.841H24v3.481h4.844a4.14 4.14 0 0 1-1.796 2.716v2.259h2.908c1.702-1.567 2.684-3.875 2.684-6.615z"
                fill="#4285F4"
              />
              <path
                d="M24 33c2.43 0 4.467-.806 5.956-2.18l-2.908-2.259c-.806.54-1.837.86-3.048.86-2.344 0-4.328-1.584-5.036-3.711h-3.007v2.332A8.997 8.997 0 0 0 24 33z"
                fill="#34A853"
              />
              <path
                d="M18.964 25.71a5.41 5.41 0 0 1-.282-1.71c0-.593.102-1.17.282-1.71v-2.332h-3.007A8.996 8.996 0 0 0 15 24c0 1.452.348 2.827.957 4.042l3.007-2.332z"
                fill="#FBBC05"
              />
              <path
                d="M24 18.58c1.321 0 2.508.454 3.44 1.345l2.582-2.58C28.463 15.891 26.426 15 24 15a8.997 8.997 0 0 0-8.043 4.958l3.007 2.332c.708-2.127 2.692-3.71 5.036-3.71z"
                fill="#EA4335"
              />
              <path
                d="M15 15h18v18H15z"
              />
            </g>
          </svg>
        </GoogleButton>
      </StyledComponent>
    </GoogleButton>
  </MockRouter>
</Provider>
`;
